lib/modules/manager/pip-compile/artifacts.ts

```diff
@@ -1,143 +1,99 @@
-import { quote, split } from 'shlex';
-import upath from 'upath';
 import { TEMPORARY_ERROR } from '../../../constants/error-messages';
 import { logger } from '../../../logger';
 import { exec } from '../../../util/exec';
 import type { ExecOptions } from '../../../util/exec/types';
 import {
   deleteLocalFile,
-  ensureCacheDir,
   readLocalFile,
   writeLocalFile,
 } from '../../../util/fs';
 import { getRepoStatus } from '../../../util/git';
-import { regEx } from '../../../util/regex';
 import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
-import {
-  allowedPipArguments,
-  constraintLineRegex,
-  getPipToolsConstraint,
-  getPythonConstraint,
-} from './common';
+import { extractHeaderCommand, getExecOptions } from './common';
 
 export function constructPipCompileCmd(
   content: string,
-  inputFileName: string,
   outputFileName: string,
+  strict: boolean = true,
 ): string {
-  const headers = constraintLineRegex.exec(content);
-  const args = ['pip-compile'];
-  if (headers?.groups) {
-    logger.debug(`Found pip-compile header: ${headers[0]}`);
-    for (const argument of split(headers.groups.arguments)) {
-      if (allowedPipArguments.includes(argument)) {
-        args.push(argument);
-      } else if (argument.startsWith('--output-file=')) {
-        const file = upath.parse(outputFileName).base;
-        if (argument !== `--output-file=${file}`) {
-          // we don't trust the user-supplied output-file argument; use our value here
-          logger.warn(
-            { argument },
-            'pip-compile was previously executed with an unexpected `--output-file` filename',
-          );
-        }
-        args.push(`--output-file=${file}`);
-      } else if (argument.startsWith('--resolver=')) {
-        const value = extractResolver(argument);
-        if (value) {
-          args.push(`--resolver=${value}`);
-        }
-      } else if (argument.startsWith('--')) {
-        logger.trace(
-          { argument },
-          'pip-compile argument is not (yet) supported',
-        );
-      } else {
-        // ignore position argument (.in file)
-      }
-    }
+  const pipCompileArgs = extractHeaderCommand(content, outputFileName);
+  if (strict && pipCompileArgs.isCustomCommand) {
+    throw new Error(
+      'Detected custom command, header modified or set by CUSTOM_COMPILE_COMMAND',
+    );
   }
-  args.push(upath.parse(inputFileName).base);
-
-  return args.map((argument) => quote(argument)).join(' ');
+  // TODO(not7cd): sanitize args that require quotes, .map((argument) => quote(argument))
+  return pipCompileArgs.argv.join(' ');
```

Comment:
```suggestion
  return pipCompileArgs.argv.map(quote).join(' ');
```
`quote` should be intelligent enough